### PR TITLE
Checking if Solc satisfies version requirements

### PR DESCRIPTION
**Pull request by Arbitrage0**
Solves issue #1365

A few things, firstly I'm new to open-source contributions and I've been dabbling with Python for a bit, so I wanted to put my skills to use on Raiden! Please do tell me what I'm doing wrong (or right). 

Secondly, this relies on the `compiler_version()` method of [_solidity.py](https://github.com/ethereum/pyethereum/blob/develop/ethereum/tools/_solidity.py) in ethereum.tools but I have a feeling that the Regex on line 127 might be a bit messed up, since I tried testing it and it doesn't come out right sometimes. I'll defer to their expertise though, might just be me. 

I've also imported the inbuilt `LooseVersion` to save time on developing a version comparison function from scratch.

### CODE REVIEW

**Comment by LefterisJP** [comment]
Hey @Arbitrage0 you have style violations. Run flake8 on the code to see what they are or just see where the travis job [failed](https://travis-ci.org/raiden-network/raiden/jobs/374673184)

**Comment by Arbitrage0** [comment]
So I think now everything is running great from my end, however the `AttributeError` that gets thrown up is because the Regex in the _solidity.py `compiler_version()` method probably isn't working (there's an extra '/' in the end which messes everything up), so it returns `None` instead of the actual version because there isn't any match, and `None` has no version attribute. 

Should I create another method from scratch?

File: raiden/blockchain/abi.py [hunk 1]
```diff
@@ -134,6 +136,13 @@ def validate_solc():
             "Make sure the solidity compiler is installed and available on your $PATH."
         )
 
+    elif LooseVersion(_solidity.compiler_version()) < LooseVersion(MIN_REQUIRED_SOLC[1:]):
+        raise RuntimeError(
+            'You are currently using the solidity compiler version {0}.\n'
```
Comment (LefterisJP) [inline raiden/blockchain/abi.py]: no need to add numbers inside the `{}` for the format string

File: raiden/blockchain/abi.py [hunk 2]
```diff
@@ -134,6 +136,13 @@ def validate_solc():
...
+            'Please upgrade to the compatible version {1}.'.format(
```
Comment (LefterisJP) [inline raiden/blockchain/abi.py]: Please upgrade to the compatible version  -> Please upgrade to the minimum compatible version

**Comment by LefterisJP** [comment]
@Arbitrage0 Hello again. Thank you for your contribution! Please address my 2 comments and also squash all your 6 commits into one.

**Comment by palango** [comment]
Hi @Arbitrage0 , we updates some code related to your change. In order to merge your PR you need to update this PR. Thank you.

**Comment by palango** [comment]
Hi @Arbitrage0 . We haven't heard back from you in two weeks so I'm closing this PR. Feel free to reopen when you revisit this.

**Pull request closed without merging**